client/src/main/java/io/apicurio/registry/rest/client/impl/RegistryClientImpl.java

```diff
@@ -0,0 +1,292 @@
+					.DELETE()
+					.build();
+
+			client.send(req, BodyHandlers.ofInputStream());
+
+		} catch (URISyntaxException | IOException | InterruptedException e) {
+			throw parseError(e);
+		}
+	}
+
+	@Override
+	public ArtifactMetaData getArtifactMetaData(String groupId, String artifactId) {
+		return null;
+	}
+
+	@Override
+	public void updateArtifactMetaData(String groupId, String artifactId, EditableMetaData data) {
+
+	}
+
+	@Override
+	public VersionMetaData getArtifactVersionMetaDataByContent(String groupId, String artifactId, Boolean canonical, InputStream data) {
+		return null;
+	}
+
+	@Override
+	public List<RuleType> listArtifactRules(String groupId, String artifactId) {
+		return null;
+	}
+
+	@Override
+	public void createArtifactRule(String groupId, String artifactId, Rule data) {
+
+	}
+
+	@Override
+	public void deleteArtifactRules(String groupId, String artifactId) {
+
+	}
+
+	@Override
+	public Rule getArtifactRuleConfig(String groupId, String artifactId, RuleType rule) {
+		return null;
+	}
+
+	@Override
+	public Rule updateArtifactRuleConfig(String groupId, String artifactId, RuleType rule, Rule data) {
+		return null;
+	}
+
+	@Override
+	public void deleteArtifactRule(String groupId, String artifactId, RuleType rule) {
+
+	}
+
+	@Override
+	public void updateArtifactState(String groupId, String artifactId, UpdateState data) {
+
+	}
+
+	@Override
+	public void testUpdateArtifact(String groupId, String artifactId, InputStream data) {
+
+	}
+
+	@Override
+	public InputStream getArtifactVersion(String groupId, String artifactId, String version) {
+		return null;
+	}
+
+	@Override
+	public VersionMetaData getArtifactVersionMetaData(String groupId, String artifactId, String version) {
+		return null;
+	}
+
+	@Override
+	public void updateArtifactVersionMetaData(String groupId, String artifactId, String version, EditableMetaData data) {
+
+	}
+
+	@Override
+	public void deleteArtifactVersionMetaData(String groupId, String artifactId, String version) {
+
+	}
+
+	@Override
+	public void updateArtifactVersionState(String groupId, String artifactId, String version, UpdateState data) {
+
+	}
+
+	@Override
+	public VersionSearchResults listArtifactVersions(String groupId, String artifactId, Integer offset, Integer limit) {
+		return null;
+	}
+
+	@Override
+	public VersionMetaData createArtifactVersion(String groupId, String artifactId, String xRegistryVersion, InputStream data) {
+		return null;
+	}
+
+	@Override
+	public ArtifactSearchResults listArtifactsInGroup(String groupId, Integer limit, Integer offset, SortOrder order, SortBy orderby) {
+
+		try {
+			final List<NameValuePair> params = List.of(new BasicNameValuePair(Parameters.LIMIT, String.valueOf(limit)),
+					new BasicNameValuePair(Parameters.OFFSET, String.valueOf(offset)),
+					new BasicNameValuePair(Parameters.SORT_ORDER, order.value()),
+					new BasicNameValuePair(Parameters.ORDER_BY, orderby.value()));
```

Comment:
Can we change this so that we just deal with `Map<String, String>` here and it gets translated into `List<NameValuePair>` inside `buildURI()`?